pkg/agent/nodeportlocal/k8s/controller.go

```diff
@@ -64,28 +117,31 @@ func podKeyFunc(pod *corev1.Pod) string {
 
 // Run starts to watch and process Pod updates for the Node where Antrea Agent is running.
 // It starts a queue and a fixed number of workers to process the objects from the queue.
-func (c *Controller) Run(stopCh <-chan struct{}) {
+func (c *NPLController) Run(stopCh <-chan struct{}) {
 	defer c.queue.ShutDown()
 	for i := 0; i < numWorkers; i++ {
 		go wait.Until(c.Worker, time.Second, stopCh)
 	}
-	c.Ctrl.Run(stopCh)
-	cache.WaitForCacheSync(stopCh, c.Ctrl.HasSynced)
+
+	go c.podInformer.Run(stopCh)
+	go c.svcInformer.Informer().Run(stopCh)
+	cache.WaitForNamedCacheSync(controllerName, stopCh, c.podInformer.HasSynced)
+	cache.WaitForNamedCacheSync(controllerName, stopCh, c.svcInformer.Informer().HasSynced)
```

Comment:
Not introduced by this PR, but normally it should wait for cachesync first then start workers, otherwise the worker may handle events without having all data synced once, leading to incomplete data, for example, listing pods that selected by a service will be incomplete and waste more CPU cycles to converge.

And waitForNamedCacheSync can take more than one cache, it should exit if it returns false which indicates the data is never synced before stopCh is closed.
```
klog.Infof("Starting %s", statusControllerName)
defer klog.Infof("Shutting down %s", statusControllerName)

if !cache.WaitForNamedCacheSync(statusControllerName, stopCh, c.podInformer.HasSynced, c.svcInformer.Informer().HasSynced) {
    return
}
for i := 0; i < numWorkers; i++ {
    go wait.Until(c.Worker, time.Second, stopCh)
}
```